pkg/server/plugin/upstreamauthority/awssecret/awssecret.go

```diff
@@ -48,6 +53,35 @@ type Configuration struct {
 	AssumeRoleARN   string `hcl:"assume_role_arn" json:"assume_role_arn"`
 }
 
+func (p *Plugin) buildConfig(coreConfig catalog.CoreConfig, hclText string, status *pluginconf.Status) *Configuration {
+	newConfig := new(Configuration)
+	if err := hcl.Decode(newConfig, hclText); err != nil {
+		status.ReportError("plugin configuration is malformed")
+		return nil
+	}
+
+	env_session_token, env_session_token_exists := os.LookupEnv("AWS_SESSION_TOKEN")
+	if newConfig.SecurityToken == "" && !env_session_token_exists {
+		status.ReportError("either the config 'secret_token' or the env variable AWS_SESSION_TOKEN must be set")
+	}
+	if newConfig.SecurityToken == "" && env_session_token == "" {
+		status.ReportError("when the config 'secret_token' is unset, the env variable AWS_SESSION_TOKEN must have a value")
+	}
+	if newConfig.SecurityToken != "" && env_session_token_exists {
+		status.ReportInfo("security token set twice, once in the config 'secret_token' and once in env variable AWS_SESSION_TOKEN, using the config file setting")
+	}
```

Comment:
Is this validation code new? I don't think it is correct either. This value is only used when attempting to use static credentials. Otherwise the AWS SDK can load credentials from many different sources.

In general, I don't think we should introduce behavior changes as part of this PR.

Reply:
In the old `validateConfig(...)` the security token was set like so

> if config.SecurityToken == "" {
>              config.SecurityToken = p.hooks.getenv("AWS_SESSION_TOKEN")
> }

And there were absolutely no errors if it was set in the config, the environment, both, or neither.

I'm restoring the no error checking behavior.